### DISCUSSION TITLE
你好，请问一下怎么修改nodeport的端口范围？

### DISCUSSION THREAD

**Discussion by chuabo** [k3s-io/k3s#5041] [Q&A] [answered]
The Service "redis-service" is invalid: spec.ports[0].nodePort: Invalid value: 36379: provided port is not in the valid range. The range of valid ports is 30000-32767

**Comment by brandond** [comment]
Check out https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver:

> --service-node-port-range <a string in the form 'N1-N2'>     Default: 30000-32767
> A port range to reserve for services with NodePort visibility. Example: '30000-32767'. Inclusive at both ends of the range.

An easier choice would be to change the port in your Service spec.

**Comment by chuabo** [comment, marked as answer]
还是自己找答案
修改/etc/systemd/system/k3s.service文件
ExecStart=/usr/local/bin/k3s server --kube-apiserver-arg service-node-port-range=30000-39999
重启 systemctl restart k3s
